frontend/src/components/JsonViewer/JsonViewerV2.tsx

```diff
@@ -0,0 +1,51 @@
+import { Box, Stack } from '@highlight-run/ui/components'
+
+import {
+	LogDetailsObject,
+	LogDetailsProps,
+	LogValue,
+} from '@/pages/LogsPage/LogsTable/LogDetails'
+
+type JsonViewerProps = {
+	attribute: LogDetailsProps['attribute']
+	allExpanded?: boolean
+	matchedAttributes?: LogDetailsProps['matchedAttributes']
+	queryParts?: LogDetailsProps['queryParts']
+}
+
+export const JsonViewerV2: React.FC<JsonViewerProps> = ({
+	attribute,
+	allExpanded = false,
+	matchedAttributes = {},
+	queryParts = [],
+}) => {
+	return (
+		<Stack py="0" gap="1">
+			{Object.entries(attribute).map(([key, value], index) => {
+				const isObject = typeof value === 'object'
+
+				return (
+					<Box key={index}>
+						{isObject ? (
+							<LogDetailsObject
+								allExpanded={allExpanded}
+								attribute={value as object}
+								label={key}
+								matchedAttributes={matchedAttributes}
+								queryParts={queryParts}
+								queryBaseKeys={[key]}
+							/>
+						) : (
+							<LogValue
+								label={key}
+								value={String(value)}
+								queryKey={key}
+								queryParts={queryParts}
+							/>
+						)}
+					</Box>
+				)
+			})}
+		</Stack>
+	)
+}
```

Comment:
Should probably move supporting components here instead of keeping them in logs.